kvbc/src/Replica.cpp

```diff
@@ -539,23 +587,34 @@ std::future<bool> Replica::getBlockAsync(uint64_t blockId, char *outBlock, uint3
 
   auto future = blocksIOWorkersPool_.async(
       [this](uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+        bool result = false;
         auto start = std::chrono::steady_clock::now();
         *outBlockSize = 0;
+        LOG_TRACE(logger, "Job Started: " << KVLOG(blockId));
 
         // getBlock will throw exception if block doesn't exist
         try {
-          getBlock(blockId, outBlock, outBlockSize);
+          result = getBlock(blockId, outBlock, outBlockSize);
         } catch (NotFoundException &ex) {
           *outBlockSize = 0;
           LOG_ERROR(logger, "Block not found:" << KVLOG(blockId));
           return false;
+        } catch (const std::runtime_error &ex) {
```

Comment:
Same as above

Reply:
I'm going to catch only the case of NotFoundException and return false. This is laterally the meaning of the bool we return in the future. Else, we have no use for the bool returned in the future, we can remove it.
I will remove all other exceptions catch.